sycl/include/sycl/esimd/detail/esimd_host_util.hpp

```diff
@@ -0,0 +1,479 @@
+//==------------- esimd_host_util.hpp DPC++ - EXPLICIT SIMD API ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Utility functions and definitions for implementing ESIMD intrinsics on host
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __SYCL_DEVICE_ONLY__
+
+#define SIMDCF_ELEMENT_SKIP(i)
+
+namespace cl {
+namespace sycl {
+namespace detail {
+namespace half_impl {
+class half;
+} // namespace half_impl
+} // namespace detail
+} // namespace sycl
+} // namespace cl
+
+using half = cl::sycl::detail::half_impl::half;
+
+#define SAT 1
```

Comment:
Such a simple macro seems very dangerous.
At least use a `constexpr` value in a namespace to avoid conflicts?
Or use very specific name?

Reply:
resolved